### PR TITLE
Adaptor Pattern Implementaion

### PR DESCRIPTION
**Pull request by Anjali-Rightpoint**
Added pods & SPM
Added classes for Room Manager, Firebase Manager, Azure Manager
Integrated LoginView and login available with Microsoft and Microsoft SSO(Firbase)
Implemented dynamic locations list

### CODE REVIEW

File: RoomFinder/RoomFinder/Location List/LocationListViewModel.swift [hunk 1]
```diff
@@ -10,16 +10,41 @@ import SwiftUI
 import Combine
 
 class LocationListViewModel: ObservableObject {
-    var dataSource: [LocationRowViewModel] = []
+    @Published var dataSource: [LocationRowViewModel] = []
     
-    let mockData: [String] = ["Chicago", "Jaipur"]
+    var mockData: [String] = []
```
Comment (carls376) [inline RoomFinder/RoomFinder/Location List/LocationListViewModel.swift:15]: #pp now that we have real data, can we change this from `mockData` to just `data`?

File: RoomFinder/RoomFinder/Authentication/LoginVIew.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import SwiftUI
+
+struct LoginView: View {
+    @ObservedObject var viewModel = LoginViewModel(loginState:false)
+    @State private var pushToNextScreen = false
+    
+    var body: some View {
+        NavigationView {
+            VStack(spacing:20) {
+                NavigationLink.init(destination: LocationListView(viewModel: LocationListViewModel()),isActive: $viewModel.isUserLoggedIn) {
```
Comment (pauluhn) [inline RoomFinder/RoomFinder/Authentication/LoginVIew.swift:18]: #? is `.init` needed?

Comment (pauluhn) [reply RoomFinder/RoomFinder/Authentication/LoginVIew.swift:18]: #? can we also move the destination into a computed property?

Comment (pauluhn) [reply RoomFinder/RoomFinder/Authentication/LoginVIew.swift:18]: #? why does `LoginView` get to create its own view model but `LocationListView` does not?

File: RoomFinder/RoomFinder/Location List/LocationListView.swift [hunk 3]
```diff
@@ -16,14 +16,16 @@ struct LocationListView: View {
     }
     
     var body: some View {
-        NavigationView {
+    //    NavigationView {
```
Comment (pauluhn) [inline RoomFinder/RoomFinder/Location List/LocationListView.swift:19]: #? do you need to keep commented out code?

File: RoomFinder/RoomFinder/NetowrkManagers/FirebaseManager.swift [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                        completion(true)
+                    }
+                    
+                }
+            }
+        }
+    }
+    
+    func fetchRooms(completion: @escaping(([RoomsListValue]) -> Void))  {
+        FirebaseSDKSettings.ref.child("value").observeSingleEvent(of: .value, with: { (snapshot) in
```
Comment (pauluhn) [inline RoomFinder/RoomFinder/NetowrkManagers/FirebaseManager.swift:39]: #? magic string?

File: RoomFinder/RoomFinder/SceneDelegate.swift [hunk 5]
```diff
@@ -20,7 +20,8 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         // This delegate does not imply the connecting scene or session are new (see `application:configurationForConnectingSceneSession` instead).
 
         // Create the SwiftUI view that provides the window contents.
-        let contentView = LocationListView(viewModel: LocationListViewModel())
+        //let contentView = LocationListView(viewModel: LocationListViewModel())
```
Comment (pauluhn) [inline RoomFinder/RoomFinder/SceneDelegate.swift:23]: #? keep commented out code?

File: RoomFinder/RoomFinder/Authentication/LoginVIew.swift (hunk 2 above)
Comment (Anjali-Rightpoint) [reply RoomFinder/RoomFinder/Authentication/LoginVIew.swift:18]: > #? why does `LoginView` get to create its own view model but `LocationListView` does not?
We have  pass a initial viewModel as LocationListView has a property for LocationListViewModel which is not an optional property.

File: RoomFinder/RoomFinder/NetowrkManagers/FirebaseManager.swift (hunk 4 above)
Comment (Anjali-Rightpoint) [reply RoomFinder/RoomFinder/NetowrkManagers/FirebaseManager.swift:39]: As of now all our locations in Firebase database are under the value node.

**Pull request merged**